pkg/clustermanager/cluster_manager.go

```diff
@@ -719,12 +719,17 @@ func (c *ClusterManager) EKSAClusterSpecChanged(ctx context.Context, clus *types
 }
 
 func compareEKSAClusterSpec(ctx context.Context, currentClusterSpec, newClusterSpec *cluster.Spec) (bool, error) {
-	currentVersionsBundle := currentClusterSpec.ControlPlaneVersionsBundle()
-	newVersionsBundle := newClusterSpec.ControlPlaneVersionsBundle()
+	for version, newVersionsBundle := range newClusterSpec.VersionsBundles {
+		currentVersionsBundle, ok := currentClusterSpec.VersionsBundles[version]
+		if !ok {
+			logger.V(3).Info("New versions bundle detected")
+			return true, nil
+		}
 
-	if currentVersionsBundle.EksD.Name != newVersionsBundle.EksD.Name {
-		logger.V(3).Info("New eks-d release detected")
-		return true, nil
+		if currentVersionsBundle.EksD.Name != newVersionsBundle.EksD.Name {
```

Comment:
this algorithm won't always work by itself. What if you had CP = 1.22 and worker1=1.21 and then you upgrade to CP=1.22 and worker1=1.22. Your algorithm will return a "no change", just because the old cluster already contained the 1.22 bundle.

I think we need to compare cp version and worker node by worker node versions.